git-link.el

```diff
@@ -797,6 +799,8 @@ With a double prefix argument invert the value of
              (message "Remote `%s' contains an unsupported URL" remote))
             ((not (functionp handler))
              (message "No handler found for %s" (car remote-info)))
+            ((equal filename "")
+             (git-link-homepage remote))
```

Comment:
Thanks for the PR —with videos! 📼

Will have an in-depth look next week (please message if you do not hear from me!) but my initial thoughts on this line is that, all things being equal, this is a reasonable default but, one upgrading from existing versions could deem this a bug: "Link missing path". But, it is also possible I'm missing some broader context here that negates this. 

There is also a plan for v1.0, where breaking changes will be make to existing defaults and this kinda thing may be more acceptable. 

Thoughts?

Reply:
This is the special case for repository root, but this also ended up discarding branch / commit information.

The main motivator for me to put the special case in is that, at least for GitHub, an empty path after /blob (`.../blob/main/`) causes a file not found error instead of redirecting to /tree (as is done for eg. `.../blob/main/filename`).

It's probably better do it properly, though, and adapt the handlers to return the right url for this.